pkg/queue/breaker.go

```diff
@@ -76,7 +88,22 @@ func (b *Breaker) Maybe(thunk func()) bool {
 	}
 }
 
-// NewSemaphore creates a semaphore with the desired maximal and initial capacity
+// UpdateConcurrency updates the maximum number of in-flight requests.
+func (b *Breaker) UpdateConcurrency(size int32) (err error) {
+	if size > 0 {
+		err = b.sem.AddCapacity(size)
+	} else {
+		err = b.sem.ReduceCapacity(-size)
+	}
+	return err
+}
```

Comment:
Should we instead surface `Add`/`Reduce` capacity to the caller?

Reply:
I don't have a strong opinion about that, but I think that it ok to leave it here to have the caller function more readable.

Reply:
Fine with me, you can simplify the function though to:

```golang
func (b *Breaker) UpdateConcurrency(size int32) error {
	if size > 0 {
		return b.sem.AddCapacity(size)
	} else {
		return b.sem.ReduceCapacity(-size)
	}
}
```